File: 1909-remove-one-element-to-make-the-array-strictly-increasing/1909-remove-one-element-to-make-the-array-strictly-increasing.py
class Solution:
    def canBeIncreasing(self, nums: List[int]) -> bool:
        prev = nums[0]
        count = 0
        for i in range(1, len(nums)):
            if nums[i] > prev:
                prev = nums[i]
            else:
                count +=1
                if count > 1:
                    return False

                if i== 1 or nums[i]> nums[i-2]:
                    prev = nums[i]
        return True

File: 1909-remove-one-element-to-make-the-array-strictly-increasing/1909-remove-one-element-to-make-the-array-strictly-increasing.java
class Solution {
    public boolean canBeIncreasing(int[] nums) {
        int count =0;
        int prev = nums[0];

        for(int i = 1; i <= nums.length-1; i++){
            if(nums[i]> prev){
                prev = nums[i];
            }else{
                count += 1;
                if(count>1){
                    return false;
                }
                if(i==1 || nums[i]> nums[i-2]){
                    prev = nums[i];
                }
            }
        }
        return true;
    }
}